testing.go

```diff
@@ -0,0 +1,8 @@
+package mux
+
+import "net/http"
+
+// TestSetURLParam set url params
```

Comment:
Could you expand on this? Why it exists, what the normal way of setting params is; URL should also be capitalized.

Reply:
Wow thanks for a fast response! 😃  I was just thinking about that myself. I'm just looking up some concrete examples for this sort of thing and then I'll reword it.

Reply:
Suggestion:

```
// TestSetURLParam sets the URL params for the given request, to be accessed via mux.Vars for testing route behaviour.
//
// This API should only be used for testing purposes; it provides a way to inject variables into the request context.
```

Reply:
I just wrote something but yours is nicer. Thanks again.